crates/extensions/c8y_http_proxy/src/actor.rs

```diff
@@ -186,32 +188,103 @@ impl C8YHttpProxyActor {
         &mut self,
         device_id: Option<&str>,
     ) -> Result<String, C8YRestError> {
-        let url_get_id = self.end_point.get_url_for_get_id(device_id);
+        let url_get_id: String = self.end_point.get_url_for_get_id(device_id);
+        let req_build_closure = |token: String| {
+            let request_internal_id = HttpRequestBuilder::get(&url_get_id);
+            request_internal_id.bearer_auth(token)
+        };
 
-        let request_internal_id = HttpRequestBuilder::get(url_get_id);
-        let res = self.execute(request_internal_id).await?;
+        let res = self.execute_to_get_internal_id(req_build_closure).await?;
```

Comment:
Why not use the `execute_retry` function itself with a closure that just ignores the `internal_id` argument? That way, the redundancy in the `execute_to_get_internal_id` function could be avoided.